firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigComponent.java

```diff
@@ -73,8 +78,11 @@ public class RemoteConfigComponent {
   @GuardedBy("this")
   private final Map<String, FirebaseRemoteConfig> frcNamespaceInstances = new HashMap<>();
 
+  private static final Map<String, FirebaseRemoteConfig> frcNamespaceInstancesStatic =
```

Comment:
IIUC this is static since it's passed to a method in the static inner class `GlobalBackgroundListener`. It would seem that both maps of instances would need to be guarded in the same way. Could `frcNamespaceInstances` just be made `static`?

Reply:
From what I've seen mixing `static` and `GuardedBy` changes what is expected be the lock, from `this` to `RemoteConfigComponent` and causes errors in the PR tests so I think it's better to keep them separate. Also, from our use case of the map, strict thread-safety is not required and we can just piggyback off of the original maps lock. It looks like RemoteConfigComponent is just adding RC instances into the map and isn't removing them or altering them in anyway.

Reply:
Given that `frcNamespaceInstancesStatic` is only being accessed through `synchronized` methods, it should be safe. That being said, there's nothing prevent this from changing in the future. What errors were tests throwing?

Reply:
It fails all the unit and instrumentation tests(:firebase-config) when they try and build the RC package. These were the errors I was seeing.

`/home/runner/work/firebase-android-sdk/firebase-android-sdk/firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigComponent.java:82: error: [GuardedBy] Invalid @GuardedBy expression: static member guarded by instance
  private static final Map<String, FirebaseRemoteConfig> frcNamespaceInstancesStatic =`

`/home/runner/work/firebase-android-sdk/firebase-android-sdk/firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigComponent.java:341: error: [GuardedBy] This access should be guarded by 'RemoteConfigComponent'; instead found: 'RemoteConfigComponent.class'
    for (FirebaseRemoteConfig frc : frcNamespaceInstancesStatic.values()) {`

`/home/runner/work/firebase-android-sdk/firebase-android-sdk/firebase-config/src/main/java/com/google/firebase/remoteconfig/RemoteConfigComponent.java:223: error: [GuardedBy] This access should be guarded by 'RemoteConfigComponent'; instead found: 'this'
      frcNamespaceInstancesStatic.put(namespace, in);`